pkg/strategy/supertrend/strategy.go

```diff
@@ -0,0 +1,482 @@
+	if base.Sign() > 0 {
+		side = types.SideTypeSell
+	}
+
+	if quantity.Compare(s.Market.MinQuantity) < 0 {
+		return fmt.Errorf("order quantity %v is too small, less than %v", quantity, s.Market.MinQuantity)
+	}
+
+	orderForm := s.GenerateOrderForm(side, quantity, types.SideEffectTypeAutoRepay)
+
+	s.Notify("Submitting %s %s order to close position by %v", s.Symbol, side.String(), percentage, orderForm)
+
+	createdOrders, err := s.session.Exchange.SubmitOrders(ctx, orderForm)
+	if err != nil {
+		log.WithError(err).Errorf("can not place position close order")
+	}
+
+	s.orderStore.Add(createdOrders...)
+
+	s.tradeCollector.Process()
+
+	return err
+}
+
+// SetupIndicators initializes indicators
+func (s *Strategy) SetupIndicators() {
+	if s.FastDEMAWindow == 0 {
+		s.FastDEMAWindow = 144
+	}
+	s.FastDEMA = &indicator.DEMA{IntervalWindow: types.IntervalWindow{Interval: s.Interval, Window: s.FastDEMAWindow}}
+
+	if s.SlowDEMAWindow == 0 {
+		s.SlowDEMAWindow = 169
+	}
+	s.SlowDEMA = &indicator.DEMA{IntervalWindow: types.IntervalWindow{Interval: s.Interval, Window: s.SlowDEMAWindow}}
+
+	if s.SuperTrend.AverageTrueRangeWindow == 0 {
+		s.SuperTrend.AverageTrueRangeWindow = 39
+	}
+	s.SuperTrend.AverageTrueRange = &indicator.ATR{IntervalWindow: types.IntervalWindow{Window: s.SuperTrend.AverageTrueRangeWindow, Interval: s.Interval}}
+	s.SuperTrend.trend = types.DirectionUp
+	if s.SuperTrend.AverageTrueRangeMultiplier == 0 {
+		s.SuperTrend.AverageTrueRangeMultiplier = 3
+	}
+}
+
+// UpdateIndicators updates indicators
+func (s *Strategy) UpdateIndicators(kline types.KLine) {
+	closePrice := kline.GetClose().Float64()
+
+	// Update indicators
+	if kline.Interval == s.FastDEMA.Interval {
+		s.FastDEMA.Update(closePrice)
+	}
+	if kline.Interval == s.SlowDEMA.Interval {
+		s.SlowDEMA.Update(closePrice)
+	}
+	if kline.Interval == s.SuperTrend.AverageTrueRange.Interval {
+		s.SuperTrend.Update(kline)
+	}
+}
+
+func (s *Strategy) GenerateOrderForm(side types.SideType, quantity fixedpoint.Value, marginOrderSideEffect types.MarginOrderSideEffectType) types.SubmitOrder {
+	orderForm := types.SubmitOrder{
+		Symbol:           s.Symbol,
+		Market:           s.Market,
+		Side:             side,
+		Type:             types.OrderTypeMarket,
+		Quantity:         quantity,
+		MarginSideEffect: marginOrderSideEffect,
+	}
+
+	return orderForm
+}
+
+// CalculateQuantity returns leveraged quantity
+func (s *Strategy) CalculateQuantity(currentPrice fixedpoint.Value) fixedpoint.Value {
+	balance, _ := s.session.GetAccount().Balance(s.Market.QuoteCurrency)
+	amountAvailable := balance.Available.Mul(fixedpoint.NewFromFloat(s.Leverage))
+	quantity := amountAvailable.Div(currentPrice)
+
+	return quantity
+}
+
+func (s *Strategy) HasTradableBase(currentPrice fixedpoint.Value) bool {
+	base := s.Position.GetBase()
+	quantity := base.Abs()
+	if quantity.Compare(s.Market.MinQuantity) > 0 && quantity.Mul(currentPrice).Compare(s.Market.MinNotional) > 0 {
+		return true
+	}
```

Comment:
you can use `func (m Market) IsDustQuantity(quantity, price fixedpoint.Value) bool {`